src/node_file.cc

```diff
@@ -3431,6 +3432,27 @@ static void CpSyncCopyDir(const FunctionCallbackInfo<Value>& args) {
   bool verbatim_symlinks = args[5]->IsTrue();
   bool preserve_timestamps = args[6]->IsTrue();
 
+  std::function<bool(std::string, std::string)> filter_fn = nullptr;
+
+  if (args[7]->IsFunction()) {
+    Local<v8::Function> args_filter_fn = args[7].As<v8::Function>();
+
+    filter_fn = [env, args_filter_fn](std::string src,
```

Comment:
If you use std::string view you will improve performance.

Reply:
great! thanks! :smile: 

Does this look good? :slightly_smiling_face: 

c9e25e8b119f47de7ba6cf750f5fbfe619c0257d